### PR TITLE
Create ImageVectorPreviewEditorProvider to preview ImageVector inside kt file

### PR DESCRIPTION
**Pull request by egorikftp**
Basic logic for parsing is done and can be reviewed 🎉
Ref: https://github.com/ComposeGears/Valkyrie/issues/132

Introduces **"components:ir"** module as "intermediate representation" for image vector.
Later will be applied to **"components:androidx-compose"** module and replace existing classes.

https://github.com/user-attachments/assets/f2fbd9a4-5580-4dba-aa1e-3a03be05ead0

TODO:
- [x] Add parsing for lazy block
- [x] Fix tests on Windows machine

### CODE REVIEW

File: components/extensions/src/main/kotlin/io/github/composegears/valkyrie/extensions/Any.kt [hunk 1]
```diff
@@ -1,3 +1,5 @@
 package io.github.composegears.valkyrie.extensions
 
 inline fun <reified T : Any> Any?.castOrNull(): T? = this as? T
+
+inline fun <reified T : Any> Any?.cast(): T = this as T
```
Comment (Goooler) [inline components/extensions/src/main/kotlin/io/github/composegears/valkyrie/extensions/Any.kt:5]: Try https://github.com/pinterest/ktlint/blob/c10637342741504690498f545cf141405cbd7244/ktlint-rule-engine-core/src/main/kotlin/com/pinterest/ktlint/rule/engine/core/util/Any.kt#L3-L13

Comment (egorikftp) [reply components/extensions/src/main/kotlin/io/github/composegears/valkyrie/extensions/Any.kt:5]: Will replace two extension

Comment (egorikftp) [reply components/extensions/src/main/kotlin/io/github/composegears/valkyrie/extensions/Any.kt]: updated

**Comment by egorikftp** [comment]
@Goooler if your have a time, could you please review? Looks like this part finalized

**Review by Goooler — APPROVED** [review]
No much experience with PSI.

**Pull request merged**